### PR TITLE
[PIR] Migrate Sparse API reshape No.1

### PR DESCRIPTION
**Pull request by epiphanyer**
### PR Category

Execute Infrastructure

### PR Types

Others

### Description

为paddle.sparse.reshape算子添加静态图模式下的单测。
issue：#64492
pcard-67164

### CODE REVIEW

File: test/legacy_test/test_sparse_reshape_op.py [hunk 1]
```diff
@@ -145,5 +145,131 @@ def test_reshape_with_zero_or_minus_one_in_new_shape(self):
         self.check_result([6, 2, 3], [-1, 1, 3], 'csr')
 
 
+devices = ['cpu']
+if paddle.device.get_device() != "cpu":
+    devices.append(paddle.device.get_device())
+
```
Comment (risemeup1) [inline test/legacy_test/test_sparse_reshape_op.py:153]: devices = []
if paddle.device.get_device() != "cpu":
    devices.append(paddle.device.get_device())
else:
    devices.append('cpu')

File: test/legacy_test/test_sparse_reshape_op.py [hunk 2]
```diff
@@ -145,5 +145,131 @@ def test_reshape_with_zero_or_minus_one_in_new_shape(self):
...
+        a sparse tensor with shape "new_shape".
+        Compare the output of paddle.reshape and the output of
+        paddle.sparse.reshape.
+        """
+        for device in devices:
+            paddle.device.set_device(device)
+            mask = paddle.randint(0, 2, x_shape)
+            while paddle.sum(mask) == 0:
+                mask = paddle.randint(0, 2, x_shape)
+            origin_data = (
```
Comment (risemeup1) [inline test/legacy_test/test_sparse_reshape_op.py:179]: 这个while循环可以加一个异常处理，避免出现死循环hang主

File: test/legacy_test/test_sparse_reshape_op.py [hunk 3]
```diff
@@ -145,5 +145,131 @@ def test_reshape_with_zero_or_minus_one_in_new_shape(self):
...
+                paddle.rand(x_shape, dtype='float32') + 1
+            ) * mask.astype('float32')
+            sparse_data = origin_data.detach().to_sparse_coo(
+                sparse_dim=len(x_shape)
+            )
+            indices_data = sparse_data.indices()
+            values_data = sparse_data.values()
```
Comment (risemeup1) [inline test/legacy_test/test_sparse_reshape_op.py]: 这三行代码简化一下吧：indices_data, values_data = origin_data.detach().to_sparse_coo(sparse_dim=len(x_shape)).indices(), origin_data.detach().to_sparse_coo(sparse_dim=len(x_shape)).values()

**Review by risemeup1 — APPROVED** [review]
LGTM

**Comment by luotao1** [comment]
hi, @epiphanyer 
- 非常感谢你对飞桨的贡献，我们正在运营一个PFCC组织，会通过定期分享技术知识与发布开发者主导任务的形式持续为飞桨做贡献，详情可见 https://github.com/luotao1 主页说明。
- 如果你对PFCC有兴趣，请发送邮件至 [ext_paddle_oss@baidu.com](mailto:ext_paddle_oss@baidu.com)，我们会邀请你加入~
